source/simulator/assemblers/advection.cc

```diff
@@ -198,6 +224,29 @@ namespace aspect
                      )
                      * JxW;
                 }
+
+              if (use_supg)
+                {
+                  for (unsigned int j=0; j<advection_dofs_per_cell; ++j)
+                    {
+                      data.local_matrix(i,j)
+                      += tau *
+                         (
+                           (current_u * (density_c_P + latent_heat_LHS)) *
+                           scratch.grad_phi_field[i] *
+                           (
+                             -time_step * conductivity * trace(scratch.hessian_phi_field[j])
```

Comment:
Is `trace(hessian)` equivalent to the laplacians of the test function? I could imagine that `get_function_laplacians` is cheaper than `get_function_hessians` if we only use the trace anyway. Or am I mistaken?

Reply:
Sadly, there is no ``fe_values[field].laplacian``, but see how you like my change I will push in a minute.